### PR TITLE
executor: add SHOW BUILTINS command

### PR DESCRIPTION
**Pull request by ekalinin**
### What problem does this PR solve? 

Add an implementation of the `SHOW BUILTINS` command.
See #12372 

### What is changed and how it works?

```sh
mysql> show builtins;

+----------------------------+
| Name                       |
+----------------------------+
| abs                        |
| acos                       |
...
| year                       |
| yearweek                   |
+----------------------------+
266 rows in set (0.01 sec)
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by c4pt0r** [comment]
Thank you for the contribution!

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -2627,6 +2627,9 @@ func buildShowSchema(s *ast.ShowStmt, isView bool) (schema *expression.Schema) {
 	case ast.ShowAnalyzeStatus:
 		names = []string{"Table_schema", "Table_name", "Partition_name", "Job_info", "Processed_rows", "Start_time", "State"}
 		ftypes = []byte{mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeLonglong, mysql.TypeDatetime, mysql.TypeVarchar}
+	case ast.ShowBuiltins:
+		names = []string{"Builtin_name"}
```
Comment (zz-jason) [inline planner/core/planbuilder.go]: how about `supported builtin functions`?

File: expression/builtin.go [hunk 2]
```diff
@@ -731,3 +732,13 @@ func IsFunctionSupported(name string) bool {
 	_, ok := funcs[name]
 	return ok
 }
+
+// GetBuiltinList returns a list of builtin functions
+func GetBuiltinList() []string {
+	res := make([]string, 0, len(funcs))
+	for funcName := range funcs {
```
Comment (zz-jason) [inline expression/builtin.go]: not all the functions in `funcs` are implemented.

Comment (ekalinin) [reply expression/builtin.go]: Could you provide an example of such function?

File: planner/core/planbuilder.go (hunk 1 above)
Comment (ekalinin) [reply planner/core/planbuilder.go]: Done

File: expression/builtin.go (hunk 2 above)
Comment (qw4990) [reply expression/builtin.go]: Only a few functions are not implemented now, for example, `builtinRowSig`.
When their evaluation methods are called, they cause a panic like `panic("xxx.evalXXX() should never be called")`.

Comment (ekalinin) [reply expression/builtin.go]: Thanks. Added an exclusion of some functions.

**Comment by AndrewDi** [comment]
@ekalinin how about implement a system view to show more detail about builtin functions?

**Pull request merged**